core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/internal/io/ChecksumInputStreamTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal.io;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static software.amazon.awssdk.http.auth.aws.internal.util.ChecksumUtil.readAll;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.http.auth.aws.internal.checksums.Crc32Checksum;
+import software.amazon.awssdk.http.auth.aws.internal.checksums.SdkChecksum;
+import software.amazon.awssdk.http.auth.aws.internal.checksums.Sha256Checksum;
+import software.amazon.awssdk.utils.BinaryUtils;
+
+public class ChecksumInputStreamTest {
+
+    @Test
+    public void test_computesCorrectSha256() {
```

Comment:
nit, we don't really use "test" as part of test name since it's unnecessary. The patterns we've established are  `methodToTest_when_expectedBehavior` or `given_when_then`